src/protocol/ipa_prf/boolean_ops/addition_sequential.rs

```diff
@@ -48,21 +44,21 @@ where
 /// adds y to x, Output has same length as x (we dont seem to need support for different length)
 /// # Errors
 /// propagates errors from multiply
-#[cfg(all(test, unit_test))]
-pub async fn integer_sat_add<C, S>(
+pub async fn integer_sat_add<C, XS, YS>(
     ctx: C,
     record_id: RecordId,
-    x: &AdditiveShare<S>,
-    y: &AdditiveShare<S>,
-) -> Result<AdditiveShare<S>, Error>
+    x: &AdditiveShare<XS>,
+    y: &AdditiveShare<YS>,
```

Comment:
Do you really need different lengths here? I didn't support different lengths since I thought that we probably don't need it.

Reply:
Yes, I think we need different lengths.

In my current test I'm having the trigger values be 3-bit numbers, with a cap of 32 (5 bits). So I'm adding a smaller 3-bit value to the running total which is a 5-bit number.